scripts/inject-deterministic.py

```diff
@@ -1,50 +1,171 @@
+        return '<script>{}</script>'.format(script)
+
+    def update_csp_script_src(self, headers, sha256):
+        """
+        Update the CSP script directives with appropriate information
+
+        Without this permissions a page with a
+        restrictive CSP will not execute injected scripts.
+
+        Note:
+        For more background information on CSP, please refer to
+        https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/
+        Content-Security-Policy/script-src
+        https://developers.google.com/web/fundamentals/security/csp/
+        """
+
+        sources = self.get_csp_script_sources(headers)
+        add_unsafe = True
+
+        for token in self.get_csp_script_sources(headers):
+            if token == "'unsafe-inline'":
+                add_unsafe = False
+                ctx.log.info("Contains unsafe-inline")
+            elif token.startswith("'sha"):
+                sources.append("'sha256-{}'".format(sha256))
+                add_unsafe = False
+                ctx.log.info("Add she hash directive")
+                break
+
+        if add_unsafe:
+            ctx.log.info("Add unsafe")
+            sources.append("'unsafe-inline'")
+
+        return "script-src {}".format(" ".join(sources))
+
+    def get_new_csp_header(self, headers, updated_csp_script):
+        """
+        get_new_csp_header generates a new header object containing
+        the updated elements from new_csp_script_directives
+        """
+
+        if updated_csp_script:
+            directives = self.get_csp_directives(headers)
+            for index, directive in enumerate(directives):
+                if directive.startswith("script-src "):
+                    directives[index] = updated_csp_script
+
+            ctx.log.info("Original Header %s \n" % headers["Content-Security-Policy"])
+            headers["Content-Security-Policy"] = "; ".join(directives)
+            ctx.log.info("Updated  Header %s \n" % headers["Content-Security-Policy"])
+
+        return headers
+
     def response(self, flow):
+
         millis = int(round(time.time() * 1000))
 
         if "content-type" in flow.response.headers:
-
             if 'text/html' in flow.response.headers["content-type"]:
                 ctx.log.info("Working on {}".format(flow.response.headers["content-type"]))
 
                 if "charset=" in flow.response.headers["content-type"]:
                     encoding = flow.response.headers["content-type"].split("charset=")[1]
                 else:
-                    encoding = "utf-8"
+                    encoding = "latin-1"
 
                 flow.response.decode()
                 html = bytes(flow.response.content).decode(encoding)
 
                 with open("scripts/catapult/deterministic.js", "r") as jsfile:
                     js = jsfile.read().replace("REPLACE_LOAD_TIMESTAMP", str(millis))
+
+                    # remove comments from js file
+                    js = re.sub(re.compile("//.*?\n"), "", js)
```

Comment:
Why is it necessary to remove the comments?

Reply:
I thought it's affecting the encoding and generating the sha.
removed